### ISSUE TITLE
Error in viewshed function

### ISSUE THREAD

**Issue by vincentwen001** [STBrinkmann/GVI#9]
Hello. When trying to reproduce the example code by pasting them into my local environment, I encountered the error as follows:

```
Error in viewshed(observer = observer, dsm_rast = DSM, dtm_rast = DEM,
               max_distance = 200, observer_height = 1.7, plot = TRUE)  :
  no slot of name "ptr" for this object of class "SpatRaster"
```
It seems there are some problems with the recent terra package. I'm not sure if it can be fixed.  My R version is 4.3.1, and 
```
    packageVersion("terra")
[1] '1.7.46'

r$> packageVersion("sf")
[1] '1.0.14'
```

Thank you very much for your help.

**Comment by hansvancalster** [comment]
Indeed, I have noticed this too. A partial fix is in #10, but that fix won't fix the problem in `GVI::viewshed()`.

**Comment by STBrinkmann** [comment]
Hi,
first of all let me apologize for my long absence. I have moved the GVI package (and others) to a new R package [CGEI](https://github.com/STBrinkmann/CGEI). This package will provide tools used in spatial epidemiology and urban planning.

The issue you have mentioned is fixed in CGEI. Also, viewshed now supports multiple observer locations, returning a list of viewsheds.
